### DISCUSSION TITLE
Issues with Link Auth Providers to an Account

### DISCUSSION THREAD

**Discussion by Avinava** [supabase/supabase#11413] [Questions]
Hi Everyone,

I was trying to implement some sort of linking with the auth providers when users try to login. 

### Use Case

In the app user will be pre provisioned / or created and once we are ready a link will be sent to the user. User will be logging in into the system using their auth provider in this case google. 

Came across this discussion https://github.com/supabase/supabase/discussions/3098, which seems to say supabase have some sort of login. I am trying to provision user using the `supabaseAdmin.auth.admin.createUser` but seems like I am missing something.

Here is what I am doing 

### Provision / Create the user

```js
const { data: auth_users, error: auth_users_error } = await supabaseAdmin.auth.admin.createUser({
     email: 'someuser@gmail.com',
     email_confirm: true,
 });
```

This should create a verified user in system, which it does.

### User logging in 

Now when the user is logging in , am trying to do something like this

```js
const { data, error } = await supabase.auth.signInWithOAuth({
    provider: 'google',
    options: {
      redirectTo: `${process.env.BASE_URL}/api/v1/auth/callback`,
    },
});
// redirect user to data.url
```

as per the https://github.com/supabase/supabase/discussions/3098 supabase should be ideally able to find the user with email and should be able to link automatically, but apparently its trying to create a dupe user and keep getting `duplicate key value violates unique constraint "users_email_partial_key"` in the auth logs.

I also tried setting the metadata but doesn't seem to work, supabase still tries to insert a new user

```js
const { data: user, error } = await supabase.auth.admin.createUser( {
  app_metadata: { provider: 'google', providers: ['google'] },
  user_metadata: {
    iss: 'https://www.googleapis.com/userinfo/v2/me',
    name: 'Some Test',
    email: 'someuser@gmail.com',
    email_verified: true,
  },
  email: '',
  email_confirm: true,
});
```

am I missing something in the process ?

**Comment by ruggi99** [comment]
Hi @Avinava 
Supabase recently applied a migration to allow partially duplicated users.
They allow duplicates only if users are SSO users, so when `is_sso_user` is `true`.
More here: https://github.com/supabase/gotrue/blob/master/migrations/20221215195500_modify_users_email_unique_index.up.sql
I think you are creating a user without `is_sso_user` set to `true.`
I don't know if you can set that property via `supabase-js` when creating a user, but with SQL sure you can

**Reply by muhammadzadeh** [reply]
Thanks, @ruggi99 
It works, but I can't find it on the `supabase-js`. and another weird thing is generated jwt token is not validatable by supabase but the token is valid.
`AuthApiError: invalid JWT: unable to parse or verify signature, token contains an invalid number of segments`

**Reply by ruggi99** [reply]
I think it's worth opening a issue to `supabase-js` if it is not possible to set `is_sso_user` property.
As said the feature is quite new so maybe they have not implemented it yet.
For the JWT I don't know what could be possibly wrong. It seems like corrupted

**Reply by kangmingtay** [reply]
Hey @ruggi99, @Avinava and @muhammadzadeh, we've identified the bug and have made the fix in gotrue in this PR: https://github.com/supabase/gotrue/pull/929

We should be deploying the fix sometime soon this week - stay tuned!

**Reply by muhammadzadeh** [reply]
@kangmingtay I tested, but it seems the problem is not fixed, Or the change not released?

**Reply by Avinava** [reply]
@kangmingtay thanks for the update, I am waiting for this to be released 👍

**Reply by ferrannp** [reply]
Can this be related https://github.com/supabase/supabase-js/issues/721 ? Still happening in the latest `supabase-js`.
